### PR TITLE
Add a comment about _typeshed to VERSIONS

### PR DESCRIPTION
**Pull request by srittau**
Support inline comments in VERSIONS

### CODE REVIEW

**Comment by Akuli** [comment]
Personally I would prefer to keep the syntax as simple as we can. Several file formats, such as `git commit` message editing or `/etc/fstab`, support comments but require the `#` to be in the start of the line.

For `_typeshed` specifically, we already have a comment in `_typeshed/__init__.pyi` explaining what it is. If that's not enough, we can move it to `_typeshed/README.md`.

**Comment by srittau** [comment]
Comments at the end of the line are a common standard. Even Python supports them. :) The reason why I want to comment on `_typeshed` specifically in `VERSIONS` is because it's also relevant here: `VERSIONS` is our "inventory" of Python's standard library packages, so it makes sense to me to clarify here that a package is not really part of Python.

**Comment by Akuli** [comment]
Seems like this needs changes in mypy

**Review by JelleZijlstra — APPROVED** [review]

File: tests/mypy_test.py [hunk 1]
```diff
@@ -75,19 +75,19 @@ def match(fn, args, exclude_list):
...
-        max_version = parse_version(m.group(3)) if m.group(3) else (99, 99)
-        result[mod] = min_version, max_version
+    with open(fname) as f:
+        for line in f:
+            # Allow having some comments or empty lines.
+            line = line.split("#")[0].strip()
+            if line == "":
+                continue
+            m = _VERSION_LINE_RE.match(line)
+            assert m, "invalid VERSIONS line :" + line
```
Comment (Akuli) [inline tests/mypy_test.py:86]: Nit: 
```suggestion
            assert m, "invalid VERSIONS line: " + line
```

**Review by Akuli — APPROVED** [review]

**Comment by srittau** [comment]
The primer reports lots of issues, similar to #5473. I'll postpone this until that PR merged cleanly.

**Comment by JelleZijlstra** [comment]
> The primer reports lots of issues

I noticed that too, seemed like something related to third-party packages. I'll look at it for a while.

**Comment by JelleZijlstra** [comment]
Looks like I broke typeshed CI with a mypy change I just merged. I'll fix it.

**Pull request merged**